### PR TITLE
Create tests

### PR DESCRIPTION
**Pull request by AIMS-Sean-Longman**
## As a contributor 
By submitting this PR I confirm I have read and understood the Github Standards document and understand the role I play in ensuring standards, security, and assurance at DfT.

## As a PR reviewer
By accepting this PR I understand my responsibilities as described in the the Github standards document.

# Proposed changes

There are a few changes here stemming from the first change of adding unit tests. This led to a slight restructuring of the package as ggplotify didn't put the grob graph back into the ggplot object how it should be. If you turned the ggplot object back into a grob, you can't find the guide-box where the legend is. This restructuring allows for unit testing as testing is done on the updateGrob function which is an internal function now and holds the functionality. This restructuring of taking the inner workings of the package and putting it in the updateGrob function is the second change.

Unit testing also showing an issue where if the key in the legend showed 2 aspects such as a scatterplot with lines on it so the key was a dot with a line through the middle, the function didn't work as the key aspects to the grob outnumbered the label aspects. I've made a change to remove this issue by looking for key-[number]-1-1 specifically rather than removing background element of the key grob.

Final change is to add a GitHub action to run the build cmd check. There is also some updates to the README file as well.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [x] Documentation Update (change to naming or other documentation)

## Points I have checked in my code

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._

- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have checked that my changes have not broken any other functionality
- [ ] I have linked to any issues this PR fixes 

## Points for review

Add checkboxes here for any specific aspects of your PR you would like to be checked in peer review. If you don't specify anything here, the reviewer will check:

- [x] All of your unit tests pass
- [x] Your code is clear and easy to understand
- [ ] You have documented any new features
- [x] They can run the code you have written
- [x] Your changes do not break any other functionality

## Security considerations

_Put an `x` in the boxes that apply. You should have considered all of these points before submitting a PR._

- [ ] My code does not contain any secrets such as passwords, API keys, etc
- [ ] I have not accidentally uploaded any data to Github
- [ ] My code does not include any personal information such as names or email addresses

## Who is reviewing this PR?

Tag your reviewer here to ensure they get a notification!
@DIPAD-Fran-Bryden

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,18 +20,7 @@ gglegendcolour::changeLegendColour(
...
-
-``` 
-scale_color_manual(
-  values = c("group 1" = "colour 1",...),
-  labels = c("group 1" = "colour 1",...)
-)
-```
+In your ggplot2 object, you must have a legend else you'll get an error. If you have multiple parts to the key in your legend, such as if your key had a line and a dot for each element in the key, the whole key for each element should be the same colour.
 
 # Example code
```
Comment (DIPAD-Fran-Bryden) [inline README.md:25]: I think the example code can be simplified (this example does work, but a much easier version works too!), just to show people it's nice and easy. E.g.:

```
library(ggplot2)

p <- ggplot(
  mtcars,
  aes(mpg, wt, colour = factor(cyl), fill = factor(cyl))
) +
  geom_point(shape = 21, alpha = 0.5, size = 2)

p <- gglegendcolour::changeLegendColour(
  x = p,
  remove_key = TRUE
)
p
```
also works nicely 😺

File: R/changeLegendColour.R [hunk 2]
```diff
@@ -18,77 +18,11 @@ changeLegendColour <- function(x,
     stop("x must be a ggplot object")
```
Comment (DIPAD-Fran-Bryden) [inline R/changeLegendColour.R:18]: Can't add the comment in the right place, but the documentation could stand to be a little more verbose here, in particular a few lines about what you can expect the legend to look like. Would also be handy to have an example or two included in the documentation (using the @example parameter), just duplicates of the ones from the readme would be fine, just easier for people to access!

**Review by DIPAD-Fran-Bryden — APPROVED** [review]
All looks great! A couple of very minor points about examples in the inline comments, nothing that needs to be actioned as a priority! All tests run fine on my machine, R CMD check --as-cran has a note that your maintainer details in the DESCRIPTION file are malformed, you may want to copy something like naptanr to get those formatted right.

File: README.md (hunk 1 above)
Comment (AIMS-Sean-Longman) [reply README.md:25]: Changed to your simplified example, also added it as an @example

**Pull request merged**